text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+  onStatusReason: StatusReason.custom('Some other reason*'),
+});
+```
+
+### Running single-container ECS workflows
+
+Batch supports ECS workflows natively. This examples creates a `JobDefinition` that runs a single container with ECS:
+
+```ts
+const jobDefn = new batch.EcsJobDefinition(this, 'JobDefn', {
+  containerDefinition: new batch.ContainerDefinition(this, 'containerDefn', {
+    image: ecs.ContainerImage.fromRegistry('public.ecr.aws/amazonlinux/amazonlinux:latest'),
+    memoryLimitMiB: 2048,
+    compatibility: Compatibility.EC2, // specify FARGATE for Fargate workflows
+  }),
+});
+```
+
+For workflows that need persistent storage, batch supports mounting `Volume`s to the container.
+You can both provision the volume and mount it to the container in a single operation:
+
+```ts
+jobDefn.addVolume({
+  name: 'myVolume',
+  efsVolumeConfiguration: {
+    fileSystem: myFileSystem
+  },
+  containerPath: '/Volumes/myVolume',
+});
+```
+
+### Running Kubernetes Workflows:
+
+Batch also supports running workflows on EKS. The following example creates a `JobDefinition` that runs on EKS:
+
+```ts
+const eksContainer = new batch.EksContainerDefinition(this, 'myEksContainer', {
+   image: ContainerImage.fromRegistry('my-registry/my-image:latest'),
+});
+
+const eksPod = new batch.EksPod(this, 'myEksPod', {
+   containers: [eksContainer],
+});
+
+new batch.EksJobDefinition(this, 'myEksJobDefn', {
+   pod: eksPod,
+});
+
+// alternative using convienience methods:
+const pod = new batch.EksPod(this, 'myEksPod');
+const jobDefn = new batch.EksJobDefinition(this, 'myEksJobDefn', {
+  pod,
+});
+
+jobDefn.addContainer('myEksContainer', { // adds to the Pod
+  image: ecs.ContainerImage.fromRegistry('my-registry/my-image:latest'),
+});
+```
+
+You can mount `Volume`s to these containers in a single operation:
+
+```ts
+jobDefn.addEmptyDirVolume({
+  name: 'emptyDir',
+  mountPath: '/Volumes/emptyDir',
+});
+jobDefn.addHostPathVolume({
+  name: 'hostPath',
+  hostPath: '/sys', 
+  mountPath: '/Volumes/hostPath',
+});
+jobDefn.addSecretVolume({
+  name: 'secret', 
+  secret: new Secret(this, 'mySecret'),
+  mountPath: '/Volumes/secret',
+});
+```
+
+Alternatively, if you prefer to add Volumes to the container without mounting them: //do we need this section? I've only implemented this to keep API-parity with our existing ECS construct>
+
+```ts
+// Creates the volumes on the Pod
+jobDefn.addEmptyDirVolume('emptyDir');
+jobDefn.addHostPathVolume('hostPath', '/sys');
+jobDefn.addSecretVolume('secret', new Secret(this, 'mySecret'));
+```
+
+You can mount these later:
+
+```ts
+// Mounts them so they are accessible to the containers
+jobDefn.addVolumeMount('emptyDir', '/Volumes/emptyDir');
+jobDefn.addVolumeMount('hostPath', '/Volumes/hostPath');
+jobDefn.addVolumeMount('secret', '/Volumes/secret');
+```
+
+### Running Distributed Workflows
+
+Some workflows benefit from parallellization and are most powerful when run in a distributed environment,
```

Comment:
Is this ultimately to say: one job in the queue can be run on multiple instances in parallel?

Reply:
Yes, exactly that